cmd/server/app/server.go

```diff
@@ -46,6 +47,30 @@ func NewProxyCommand(p *Proxy, o *options.ProxyRunOptions) *cobra.Command {
 	return cmd
 }
 
+func tlsCipherSuites(cipherNames []string) ([]uint16, error) {
+	// return nil, so use default cipher list
+	if len(cipherNames) == 0 {
+		return nil, nil
+	}
+
+	// acceptedCiphers are all the ciphers supported by the tls package
+	acceptedCiphers := make(map[string]uint16, len(tls.CipherSuites()))
+	for _, v := range tls.CipherSuites() {
+		acceptedCiphers[v.Name] = v.ID
+	}
+
+	// compare user given ciphers and accepted ciphers, and collect them
+	ciphersIntSlice := make([]uint16, 0)
+	for _, cipher := range cipherNames {
+		intValue, ok := acceptedCiphers[cipher]
+		if !ok {
+			return nil, fmt.Errorf("cipher suite %s not supported, doesn't exist or considered as insecure", cipher)
```

Comment:
Any chance this could be moved into the validation functions?

Reply:
moved into the validation functions